File: Retos/Reto #18 - WEB SCRAPING [Difícil]/ruby/test0n3.rb
# Ruby app, requires to have installed the following gems:
# - URI
# - net/http
# - nokogiri

# frozen_string_literal: true

require 'uri'
require 'net/http'
require 'nokogiri'

# class GetSchedule, get on date schedule
class ScheduleHolaMundoDay
  HOLAMUNDO_URL = 'https://holamundo.day/'

  def pick_data
    uri = URI(HOLAMUNDO_URL)
    res = Net::HTTP.get_response(uri)
    show_schedule(res.body) if res.is_a?(Net::HTTPSuccess)
  end

  def show_schedule(response)
    doc = Nokogiri::HTML(response)
    schedules = find_schedules(doc)[0]

    clean_html_snip(schedules).drop(8).map(&:content)
  end

  private

  def find_schedules(html_body)
    html_body.css('article.notion-page-content-inner').select do |article|
      article.content.downcase.match?(/(agenda 8 de mayo)/)
    end
  end

  def clean_html_snip(html_snip)
    html_snip.children.select { |elem| elem.name == 'h1' || elem.name == 'blockquote' }
  end
end

puts ScheduleHolaMundoDay.new.pick_data

File: Retos/Reto #18 - WEB SCRAPING [Difícil]/javascript/test0n3.js
// JavaScript App that can only be applied through Browser console
// Aplicación en JavaScript que solo se puede aplicar en la consola del navegador

let articles = document.getElementsByClassName("notion-page-content-inner");
let expectedArticle = "";
for (let item of articles) {
  if (item.innerHTML.toLowerCase().match(/(agenda 8 de mayo)/g))
    expectedArticle = item.children;
}

let schedule = [];
for (let item in expectedArticle) {
  if (
    item > 11 &&
    (expectedArticle[item].tagName == "H1" ||
      expectedArticle[item].tagName == "BLOCKQUOTE")
  ) {
    schedule.push(expectedArticle[item].textContent);
    console.log(expectedArticle[item].textContent);
  }
}
